src/main/java/controllers/ConsoleInputController.java

```diff
@@ -0,0 +1,124 @@
+package controllers;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import models.data.Project;
+import repositories.ProjectRepository;
+import util.log.DukeLogger;
+import views.CLIView;
+
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class ConsoleInputController implements IController {
+
+    private CLIView consoleView;
+    private ProjectRepository projectRepository;
+    private ProjectInputController projectInputController;
+    private String filePath = System.getProperty("user.dir") + "/savedProjects.json";
+
+    /**
+     * Constructor.
+     * @param view : takes in a View model, in this case a command line view.
+     */
+    public ConsoleInputController(CLIView view) {
+        this.consoleView = view;
+        loadProjectsData();
+        this.projectInputController = new ProjectInputController(this.consoleView, this.projectRepository);
+    }
+
+    /**
+     * Method that is called upon receiving commands from CLI.
+     * @param input : Input typed by user into CLI
+     */
+    @Override
+    public void onCommandReceived(String input) {
+        DukeLogger.logInfo(ConsoleInputController.class, "input:'" + input + "'");
+        Scanner inputReader = new Scanner(input);
+        String command = inputReader.next();
+
+        switch (command) {
+        case "bye":
+            consoleView.end();
+            break;
+        case "list":
```

Comment:
Such back and forth calling is not the most elegant way approach this problem. Consider having dedicated controllers for business logic related calls.